server/src/test/java/io/littlehorse/common/model/getable/core/wfrun/subnoderun/WorkflowEventRunTest.java

```diff
@@ -0,0 +1,80 @@
+package io.littlehorse.common.model.getable.core.wfrun.subnoderun;
+
+import io.littlehorse.TestUtil;
+import io.littlehorse.common.LHConstants;
+import io.littlehorse.common.model.corecommand.CommandModel;
+import io.littlehorse.common.model.corecommand.subcommand.StopWfRunRequestModel;
+import io.littlehorse.common.model.getable.core.events.WorkflowEventModel;
+import io.littlehorse.common.model.getable.core.noderun.NodeRunModel;
+import io.littlehorse.common.model.getable.core.wfrun.ThreadRunModel;
+import io.littlehorse.common.model.getable.global.wfspec.node.NodeModel;
+import io.littlehorse.common.model.getable.objectId.NodeRunIdModel;
+import io.littlehorse.common.model.getable.objectId.WfRunIdModel;
+import io.littlehorse.common.model.getable.objectId.WorkflowEventDefIdModel;
+import io.littlehorse.common.model.getable.objectId.WorkflowEventIdModel;
+import io.littlehorse.common.proto.Command;
+import io.littlehorse.sdk.common.proto.LHStatus;
+import io.littlehorse.server.TestProcessorExecutionContext;
+import io.littlehorse.server.streams.topology.core.CommandProcessorOutput;
+import io.littlehorse.server.streams.util.HeadersUtil;
+import java.util.Date;
+import java.util.UUID;
+import org.apache.kafka.common.header.Headers;
+import org.apache.kafka.streams.processor.api.MockProcessorContext;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Answers;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class WorkflowEventRunTest {
+
+    private final MockProcessorContext<String, CommandProcessorOutput> mockProcessor = new MockProcessorContext<>();
+    private final Headers metadata =
+            HeadersUtil.metadataHeadersFor(LHConstants.DEFAULT_TENANT, LHConstants.ANONYMOUS_PRINCIPAL);
+    private final Command dummyCommand = buildCommand();
+    private final TestProcessorExecutionContext testProcessorContext =
+            TestProcessorExecutionContext.create(dummyCommand, metadata, mockProcessor);
+
+    @Test
+    public void shouldStoreNode() throws Exception {
```

Comment:
usually tests help to understand the code, but firts the test has to be easy to read, is it possible to refactor this?

Reply:
We need a better way to test subnodes run. I'll add a ticket to create a generic way to test subnodes.